Makefile

```diff
@@ -67,11 +73,13 @@ test-coverage:
 	sleep 10
 	docker compose -f dev/docker-compose-integration.yml cp ./dev/provision.py spark-iceberg:/opt/spark/provision.py
 	docker compose -f dev/docker-compose-integration.yml exec -T spark-iceberg ipython ./provision.py
-	poetry run coverage run --source=pyiceberg/ -m pytest tests/ ${PYTEST_ARGS}
+	poetry run coverage run --source=pyiceberg/ -m pytest tests/ -m integration ${PYTEST_ARGS}
```

Comment:
If we set the `--data-file` argument to a specific value for the unit tests (`.coverage.unit`), and then a different one for the integration test (`.coverage.integration`), then we could `coverage combine` these data files and then produce the `coverage report` together.